### ISSUE TITLE
CommaDelimitedList returns E2015 Default should be allowed by AllowedPattern

### ISSUE THREAD

**Issue by andyloughran** [aws-cloudformation/cfn-lint#2686]
### CloudFormation Lint Version

0.72.2

### What operating system are you using?

Mac

### Describe the bug

When running AllowedPattern with CommaDelimitedList, when the Default is set to a comma-delimited-value, the Default isn't evaluated per comma, but as a whole - so fails.

### Expected behavior

The Default should be split by comma, and then evaluted against the AllowedPattern.

### Reproduction template

```yaml
AWSTemplateFormatVersion: "2010-09-09"
Parameters:
 AWSOrganizationId:
    Description: "Comma-separated IDs of AWS Organizations where this account and the target pipeline account are members."
    Type: "CommaDelimitedList"
    Default: "o-sdfs234sf,o-fdsd243s"
    AllowedPattern: "^[a-z0-9-]+$"

**Comment by andyloughran** [comment]
This was fixed in version 0.77 🎉
